sphinx/application.py

```diff
@@ -403,21 +392,14 @@ def warn(self, message, location=None, prefix='WARNING: ',
            :meth:`.BuildEnvironment.warn` since that will collect all
            warnings during parsing for later output.
         """
-        if is_suppressed_warning(type, subtype, self.config.suppress_warnings):
-            return
+        if prefix:
+            warnings.warn('prefix option of warn() is now deprecated.',
+                          RemovedInSphinx17Warning)
+        if colorfunc:
+            warnings.warn('colorfunc option of warn() is now deprecated.',
+                          RemovedInSphinx17Warning)
 
-        if isinstance(location, tuple):
-            docname, lineno = location
-            if docname:
-                location = '%s:%s' % (self.env.doc2path(docname), lineno or '')
-            else:
-                location = None
-        warntext = location and '%s: %s%s\n' % (location, prefix, message) or \
-            '%s%s\n' % (prefix, message)
-        if self.warningiserror:
-            raise SphinxWarning(warntext)
-        self._warncount += 1
-        self._log(colorfunc(warntext), self._warning, True)
+        logger.warning(message, type=type, subtype=subtype, location=location)
```

Comment:
When will we deprecate this app.warn function? The current implementation does not seem to warn deprecatoin yet. I think it is appropriate to set RemovedInSphinx**18**Warning(PendingDeprecationWarning). (OR maybe RemovedInSphinx20Warning?)